### DISCUSSION TITLE
How to use ICollectionFixture with a Fixture with Constructor that have parameters?

### DISCUSSION THREAD

**Discussion by orney21d** [xunit/xunit#2459] [Question] [answered]
Hi ,

We have an scenario where we are using a ICollectionFixture  like below :

```
 public CollectionFeatureA(
            Dep1 dep1,
            ILogger<CollectionFeatureA> logger)
        {
            _dep1 = dep1;
            _logger = logger;
        }
```

then we define the collectionFixture to use:

```

[CollectionDefinition("CollectionName")]
    public class CollectionDefinitionClass
        : ICollectionFixture<CollectionFeatureA>
```

Then in the class using the global shared type we have something like this:

```
[Collection("CollectionName")]
    public class TestedClass
    {
        private readonly CollectionFeatureA _dependency;

        public TestedClass(
            CollectionFeatureA  dependency
           )
        {
            _dependency = dependency;
        }
        

     ..... TestMethods_Using_Dependency......(){
     }
    }
```

The fact is when trying to build and execute the integration test in the CD pipeline, we are getting the error :

_Collection fixture type 'xxxx.**CollectionFeatureA**' had one or more unresolved constructor arguments: Dep1 dep1,  ILogger`1 logger_

Any advise how to use ICollectionFixture with a constructor with parameters and inject them to use it properly ??

Thanks in advance.

**Comment by bartelink** [comment]
In V2, Collection Fixtures cannot have arguments - its not a generic DI mechanism. (See recent discussions for similar with more detail)

**Comment by orney21d** [comment, marked as answer]
Thanks for the feedback ;)
